File: src/py/gaussian.py

from math import exp, pi, sqrt


def gaussianCDF(mean, standardDev, x):
	denominator = 1
	sum = 0
	assert standardDev > 0
	z = (x - mean) / standardDev
	phi = exp(-(z ** 2) / 2) / sqrt(2 * pi)
	for i in range(1, 100, 2):
		denominator *= i
		sum += z ** i / denominator
	result = 0.5 + phi * sum
	return result

File: src/java/Gaussian.java
public class Gaussian {
  public static double gaussianCDF(double mean, double standardDev, double x) {
    double phi, result, z, denominator = 1,
        sum = 0;
    int i;
    assert (standardDev > 0);
    z = (x - mean) / standardDev;
    phi = Math.exp(-Math.pow(z, 2) / 2) / Math.sqrt(2 * Math.PI);
    for (i = 1; i <= 100; i += 2) {
      denominator *= i;
      sum += Math.pow(z, i) / denominator;
    }
    result = 0.5 + phi * sum;
    return result;
  }
}